### ISSUE TITLE
'ErrorBoundary' cannot be used as a JSX component.

### ISSUE THREAD

**Issue by KubaJastrz** [bvaughn/react-error-boundary#145]
- `react-error-boundary` version: 4.0.8
- `node` version: latest
- `npm` version: latest

Relevant code or config

```js
import { ErrorBoundary } from 'react-error-boundary';

function App() {
  return (
    <ErrorBoundary fallback={<div>Something went wrong</div>}>
      <h1>Hello</h1>
    </ErrorBoundary>
  );
}
```

What you did: Installed this package and started using it.

What happened: The app stopped compiling.

Reproduction repository: https://stackblitz.com/edit/stackblitz-starters-d6a6ck?file=src%2FApp.tsx

Problem description: Wrong types?

Suggested solution: IDK, but I've seen this is a recurring issue here.

**Comment by fh-igor-serko** [comment]
@KubaJastrz the issue is specifically with @types/react. In my case I had several packages installing different versions of @types/react, but there was one which installed the latest version even though I was on React 17 and this is what caused the problem.

I had to use the `overrides` option in package.json (or `resolutions` if using Yarn) to specify a really specific version of @types/react

So if using npm I added this to `package.json`:

```
{
  ...
  "overrides": {
    "@types/react": "17.0.60"
  },
  ...
}
```

If using Yarn:

```
{
  ...
  "resolutions": {
    "@types/react": "17.0.60"
  },
  ...
}
```

Caveat, it really depends on your dependencies, so additional tweaks may be necessary.

**Comment by bvaughn** [comment]
I'm going to push out a release this morning that just changes the `children` function return type to `ReactElement` too (instead of `ReactNode`). I _think_ this should help as well.

**Comment by bvaughn** [comment]
Sorry, my previous comment was mistakenly mixing up two of my projects (this one and auto-sizer).

**Comment by bvaughn** [comment]
I'm going to close this issue and add a section to the FAQ about it. Thanks so much @fh-igor-serko 🙇🏼

**Comment by KubaJastrz** [comment]
So `@types/react` is broken above the version 17.0.60? Do the maintainers of the package know about it?

**Comment by bvaughn** [comment]
> So @types/react is broken above the version 17.0.60

No. react and @types/react need to be in sync (at least with the major versions). If you're using react@18, make sure to also use @types/react@18, etc.

**Comment by KubaJastrz** [comment]
I think this is a wrong observation. The versions are matched in my example. 
![image](https://github.com/bvaughn/react-error-boundary/assets/6443113/c616f0fd-7d87-496b-8b98-d8fad4d4c91e)

I made another test case with class components which works fine: https://stackblitz.com/edit/stackblitz-starters-cekd8h?file=src%2FApp.tsx

**Comment by KubaJastrz** [comment]
@bvaughn in fact, if you make a 1:1 reproduction with the code from this repo, you will see it being broken again: https://stackblitz.com/edit/stackblitz-starters-gch3dl?file=src%2FApp.tsx

https://github.com/bvaughn/react-error-boundary/blob/8fafb7728b74c449b8c9438ce4885ba678fbe778/src/ErrorBoundary.ts#L105-L115

The key thing that breaks the types is `as ReactElement` in your code.

**Comment by bvaughn** [comment]
Fix released in 4.0.9
